test/test_infer_util.py

```diff
@@ -40,26 +41,74 @@ def model(data=None):
     return model, data, true_probs
 
 
-def test_predictive():
+@pytest.mark.parametrize('parallel', [True, False])
+def test_predictive(parallel):
     model, data, true_probs = beta_bernoulli()
     mcmc = MCMC(NUTS(model), num_warmup=100, num_samples=100)
     mcmc.run(random.PRNGKey(0), data)
     samples = mcmc.get_samples()
-    predictive_samples = predictive(random.PRNGKey(1), model, samples)
+    predictive = Predictive(model, samples, parallel=parallel)
+    predictive_samples = predictive.get_samples(random.PRNGKey(1))
     assert predictive_samples.keys() == {"obs"}
 
-    predictive_samples = predictive(random.PRNGKey(1), model, samples,
-                                    return_sites=["beta", "obs"])
+    predictive.return_sites = ["beta", "obs"]
+    predictive_samples = predictive.get_samples(random.PRNGKey(1))
     # check shapes
     assert predictive_samples["beta"].shape == (100,) + true_probs.shape
     assert predictive_samples["obs"].shape == (100,) + data.shape
     # check sample mean
     assert_allclose(predictive_samples["obs"].reshape((-1,) + true_probs.shape).mean(0), true_probs, rtol=0.1)
 
 
+def test_predictive_with_guide():
```

Comment:
We should add a test with an autoguide. I am not fully sure if this will work correctly with autoguides, and we may need something like `Predictive(model, poutine.block(guide, hide=[list of obs sites])`.

Reply:
For autoguide, I used its `sample_posterior` method without any problem yet. This interface (Predictive with `guide`) might not generate correct sample sites (because AutoGuide's `guide` generates samples in base distribution's domain). >"<

Reply:
Hmm..that's not ideal. We should at the very least document that, or try to make the utility function work in either case (a hacky way would be to check if the guide is an Autoguide and call `sample_posterior` but its annoying to have utility function take on a contrib dependency).

Reply:
I think we should add a note to AutoContinuous docs.